symengine/visitor.h

```diff
@@ -72,7 +72,7 @@ void postorder_traversal_stop(const Basic &b, StopVisitor &v);
 class HasSymbolVisitor : public BaseVisitor<HasSymbolVisitor, StopVisitor>
 {
 protected:
-    RCP<const Symbol> x_;
+    const Symbol* x_;
```

Comment:
`Ptr` cannot have `nullptr` so it cannot have state. However, you have two distinct states with visitors: unapplied and applied. In other words, the `Ptr<const Symbol>` is very much like a reference that must be immediately initialized.